packages/amplify_storage_s3/ios/Classes/AmplifyStorageOperations.swift

```diff
@@ -141,16 +144,55 @@ public class AmplifyStorageOperations {
-        var errorDetails: [String: Any] = [:]
-        var platformErrorDetails: [String: String] = [:]
-        
-        platformErrorDetails["platform"] = "iOS"
-        platformErrorDetails["localalizedError"] = localizedError
-        platformErrorDetails["recoverySuggestion"] = recoverySuggestion
-        
-        errorDetails["PLATFORM_EXCEPTIONS"] = platformErrorDetails
-        
-        flutterResult(FlutterError(code: "AmplifyException", message: msg, details: errorDetails))
+    private static func prepareError(flutterResult: @escaping FlutterResult, storageError: StorageError) {
+        let details: Dictionary<String, String> = createSerializedError(storageError: storageError);
+        logErrorContents(error: storageError)
+           ErrorUtil.postErrorToFlutterChannel(
+               result: flutterResult,
+               errorCode: "StorageException",
+               details: details)
+    }
+    
+    private static func prepareGenericError(flutterResult: @escaping FlutterResult, error: Error) {
+        logErrorContents(error: error)
+        var underlyingErrorString: String = error.localizedDescription
+
+        if (error is AmplifyFlutterValidationException) {
+                    let e = error as! AmplifyFlutterValidationException
+                    underlyingErrorString = e.errorDescription
+                }
+                let serializedErrror = createSerializedError(
+                    message: "An unexpected error has occurred",
+                    recoverySuggestion: "See iOS logs for details",
+                    underlyingError: underlyingErrorString)
+
+                ErrorUtil.postErrorToFlutterChannel(
+                    result: flutterResult,
+                    errorCode: "StorageException",
+                    details: serializedErrror)
+            
+
+    }
+
+   static func createSerializedError(storageError: StorageError)  -> Dictionary<String, String> {
+        var serializedException: Dictionary<String, String> = [:]
+        serializedException["message"] = storageError.errorDescription
+        serializedException["recoverySuggestion"] = storageError.recoverySuggestion
+        serializedException["underlyingException"] = storageError.underlyingError.debugDescription
+        return serializedException
+    }
+
+    static func createSerializedError(message: String, recoverySuggestion: String, underlyingError: String)  -> Dictionary<String, String> {
+        var serializedException: Dictionary<String, String> = [:]
+        serializedException["message"] = message
+        serializedException["recoverySuggestion"] = recoverySuggestion
+        serializedException["underlyingException"] = underlyingError
+        return serializedException
+    }
+    
+    static func logErrorContents(error: Error) {
+        log.error(error: error)
     }
 }
+
+extension AmplifyStorageOperations: DefaultLogger { }
```

Comment:
Needed?

Reply:
Yes, I believe this is needed to enable the logger.  Worked with Jithin on this in auth.